src/main/java/com/gmail/val59000mc/maploader/CaveOresOnlyPopulator.java

```diff
@@ -60,52 +79,70 @@ private void scanChunk(Chunk chunk){
         }
     }
 
-    private static class Vein{
+    private class Vein{
         private final Set<Block> ores;
-
-        private Vein(){
-            ores = new HashSet<>();
+        private final Chunk chunk;
+        private final Material type;
+        private final Block startBlock;
+
+        private Vein(Chunk chunk, Block startBlock){
+            this.ores = new HashSet<>();
+            this.chunk = chunk;
+            this.type = startBlock.getType();
+            this.startBlock = startBlock;
         }
 
-        private void process(Block startBlock){
-            getVeinBlocks(startBlock, startBlock.getType(), 2, 10);
+        private void process(){
+            getVeinBlocks(startBlock);
         }
 
-        private void getVeinBlocks(Block block, Material type, int i, int maxBlocks){
-            if (maxBlocks == 0) return;
-
-            if (block.getType() == UniversalMaterial.GLOWING_REDSTONE_ORE.getType()){
-                block.setType(Material.REDSTONE_ORE);
-            }
-
-            if (block.getType() == type && !ores.contains(block)){
-                ores.add(block);
-                i = 2;
-            }else {
-                i--;
-            }
-            if (i > 0){
-                for (BlockFace face : BLOCK_FACES) {
-                    getVeinBlocks(block.getRelative(face), type, i, maxBlocks-1);
-                }
+        private void getVeinBlocks(Block block){
+        	
+        	int relX = block.getX() & 0x0000000F;
+        	int relY = block.getY();
+        	int relZ = block.getZ() & 0x0000000F;
```

Comment:
Just out of intrest, why did you use bitwise operators instead of
`block.getX() % 16`

Reply:
Yeah you're right, I was used last months to design hardware in VHDL. Using %16 and /16 is of course better in terms of readability.